### DISCUSSION TITLE
Using DISTINCT with additional columns, but only select the entity

### DISCUSSION THREAD

**Discussion by MangalaUral** [sqlalchemy/sqlalchemy#12155] [Usage Questions]
We are in the process of upgrading from 1.3 to 2.0. We have multiple queries where distinct and order by is used. In earlier version, even though the column is not added in the select list, implicilty sqlalchemy was doing it. But after upgrading it is throwing error. Need a common solution to add this we have more than 400 queries to be changed.

**Comment by zzzeek** [comment]
hi -

I have no way to respond to this without illustration of complete round trip examples of what these queries look like, what database is used, what results are desired.

**Comment by zzzeek** [comment]
There is a migration note for something that might be what you're referring to here which you can see at https://docs.sqlalchemy.org/en/20/changelog/migration_20.html#using-distinct-with-additional-columns-but-only-select-the-entity .  this might be what you're looking at, this has migration notes

also I  would strongly recommend getting your 1.3 code to run on 1.4 first, then run in deprecations mode as is documented at https://docs.sqlalchemy.org/en/20/changelog/migration_20.html#x-2-x-migration-overview
